app.js

```diff
@@ -1,23 +1,37 @@
 const express = require("express");
-const { getTopics } = require("./controllers/news-controllers");
+const { getTopics } = require("./controllers/topics.controller");
+const { getArticleByID } = require("./controllers/articles.controller");
 
 const app = express();
 
 app.get("/api/topics", getTopics);
 
-//custom errors
+app.get("/api/articles/:articleid", getArticleByID);
+
 app.all("/*", (req, res, next) => {
-  res.status(404).send({ msg: "not found", detail: "inavlid endpoint/method" });
+  res.status(404).send({ msg: "not found", detail: "invalid endpoint/method" });
+});
+
+//psql errors
+app.use((err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({
+      msg: "bad request",
+      detail: "invalid data type, please enter a valid data type",
+    });
+  } else next(err);
 });
 
+//custom error handler
 app.use((err, req, res, next) => {
-  if (err.status) {
+  if (err.status && err.msg && err.detail) {
     res.status(err.status).send({ msg: err.msg, detail: err.detail });
   }
```

Comment:
I dont know if it will make a difference long term with issues but i would have an else here too since you are doing it on your previous error handler. Consistency is really important with readability. This reads that you want to send an error msg and then also pass it on to the next error handler which may cause issues with double sending responses

Reply:
Awesome will do.